### PR TITLE
HYDRAN-200: Implement precheck resource

### PR DESCRIPTION
**Pull request by michaelcarlberg**
## Types of changes

- [ ] Bug fix
- [x] New feature
- [ ] Removed feature
- [ ] Code style update (formatting etc.)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Content/Data

## Does this PR introduce a breaking change?
- [ ] Yes (I have stepped the version number accordingly)
- [x] No
      
## Checklist:

- [x] My code follows the code style of this project.
- [ ] I have updated the documentation accordingly (if applicable).
- [x] I have added/updated tests to cover my changes (if applicable).

### CODE REVIEW

File: src/test/java/se/sundsvall/postportalservice/api/PrecheckResourceTest.java [hunk 1]
```diff
@@ -1,43 +1,67 @@
...
-			.bodyValue(createValidPrecheckRequest())
+			.uri("/{municipalityId}/{departmentId}/precheck", MUNICIPALITY_ID, DEPARTMENT_ID)
+			.contentType(APPLICATION_JSON)
+			.accept(APPLICATION_JSON)
+			.bodyValue(request)
 			.exchange()
-			.expectStatus().isEqualTo(HttpStatus.NOT_IMPLEMENTED);
+			.expectStatus().isOk()
+			.expectBody().jsonPath("$.recipients.length()").isEqualTo(2);
 	}
```
Comment (josc04) [inline src/test/java/se/sundsvall/postportalservice/api/PrecheckResourceTest.java:53]: Verify that mock has been called (I know it is done in the background, but it is nice to have it visible in the test also)?

File: src/test/java/se/sundsvall/postportalservice/api/PrecheckResourceTest.java [hunk 2]
```diff
@@ -1,43 +1,67 @@
...
 		webTestClient.post()
-			.uri(uriBuilder -> uriBuilder.replacePath("/{municipalityId}/precheck")
-				.build(INVALID_MUNICIPALITY_ID))
-			.bodyValue(createValidPrecheckRequest())
+			.uri("/{municipalityId}/{departmentId}/precheck", INVALID_MUNICIPALITY_ID, DEPARTMENT_ID)
+			.contentType(APPLICATION_JSON)
+			.accept(APPLICATION_JSON)
+			.bodyValue(request)
 			.exchange()
 			.expectStatus().isBadRequest();
```
Comment (josc04) [inline src/test/java/se/sundsvall/postportalservice/api/PrecheckResourceTest.java:65]: Verify no interactions with mock

File: src/test/java/se/sundsvall/postportalservice/service/PrecheckServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+
+	@Mock
+	PrecheckMapper precheckMapper;
+
+	@InjectMocks
+	PrecheckService precheckService;
+
+	@AfterEach
+	void noMoreInteractions() {
+		verifyNoMoreInteractions(citizenIntegrationMock);
```
Comment (josc04) [inline src/test/java/se/sundsvall/postportalservice/service/PrecheckServiceTest.java:65]: You could remove rows 66 and 67 by putting multiple mocks into this verification, i.e. 
```
verifyNoMoreInteractions(
   citizenIntegrationMock, 
   messagingSettingsIntegrationMock, 
   messagingIntegrationMock); 
```
to save some bytes 🙂

File: src/test/java/se/sundsvall/postportalservice/service/mapper/PrecheckMapperTest.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+	private final PrecheckMapper precheckMapper = new PrecheckMapper();
+
+	@Test
+	void toFailureByPersonId() {
+		final var personGuidBatchOK = createPersonGuidBatch("191111111111", true, null, UUID.randomUUID());
+		final var personGuidBatchFail1 = createPersonGuidBatch("192222222222", false, "some random error", null);
+		final var personGuidBatchFail2 = createPersonGuidBatch("193333333333", false, null, null);
+		final var personGuidBatches = List.of(personGuidBatchOK, personGuidBatchFail1, personGuidBatchFail2);
+
+		final var result = precheckMapper.toFailureByPersonId(personGuidBatches);
```
Comment (josc04) [inline src/test/java/se/sundsvall/postportalservice/service/mapper/PrecheckMapperTest.java:28]: Add test when inparameter is null (comment goes for all tested methods in mapper)

**Review by josc04 — APPROVED** [review]

**Review by josc04 — APPROVED** [review]
